models/users.js

```diff
@@ -0,0 +1,53 @@
+const mongoose = require( 'mongoose' );
+const Schema = mongoose.Schema;
+const bcrypt = require( 'bcrypt-nodejs' );
+
+const UserSchema = new Schema({
+<<<<<<< HEAD
+   name: String,
+=======
+>>>>>>> 6a1c686e0138cc4f22e96c08935f90a95ee3802f
+   email: { type: String, required: true, unique: true },
+   password: { type: String, required: true },
+   token: { type: String },
+   token_expires: { type: Date },
+   created_at: Date,
+   updated_at: Date
+});
+
+UserSchema.pre( 'save', function( next ){
+  const user = this
+  const SALT_FACTOR = 5
+
+  if ( !user.isModified( 'password' )){
+    return next()
+  }
+
+  bcrypt.genSalt( SALT_FACTOR, ( err, salt ) => {
+    if ( err ) return next( err )
+
+    bcrypt.hash( user.password, salt, null, ( err, hash ) => {
+      if ( err ) return next(err)
```

Comment:
Please omit spaces between keywords and opening parens. No single line if statements.